crawl-ref/source/mon-act.cc

```diff
@@ -886,6 +886,8 @@ static bool _handle_swoop_or_flank(monster& mons)
             }
         }
         mons.move_to_pos(tracer.path_taken[j+1]);
+        // Apply traps and other location effects to the monster.
+        mons.apply_location_effects(mons.pos());
```

Comment:
It looks like `apply_location_effects` wants the monster's old position, not its new one. Could you fix this?